gemma-teixido/challenge/to-do-list-redux/src/components/ToDoList/index.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+  addTask, deleteTask, loadTasks, doneTask
+} from '../../redux/actions/actionCreator';
+
+import './ToDoList.css';
+
+function ToDoList({ tasks, dispatch }) {
+  useEffect(() => {
+    if (!tasks.length) dispatch(loadTasks());
+  }, []);
+
+  const [newestTask, setNewestTask] = useState(null);
+
+  const handleDelete = (taskId) => {
+    dispatch(deleteTask(taskId));
+  };
+
+  function handleDone(taskId) {
+    dispatch(doneTask(taskId));
+  }
+
+  const handleAddTask = () => {
+    let newId = 1;
+    if (tasks.length !== 0) newId = tasks[tasks.length - 1].id + 1;
+    const task = {
+      id: newId,
+      name: newestTask
+    };
+    dispatch(addTask(task));
+  };
+
+  function taskInputCreation(event) {
+    setNewestTask(event.target.value);
+  }
+
+  return (
+
+    <section>
+      <div className="inputTask">
+        <label htmlFor="taskInputCreation">
+          <input onChange={taskInputCreation} placeholder="New task..." />
+          <button type="button" className="addTask" onClick={() => handleAddTask()}> </button>
+        </label>
+      </div>
+      <ul>
+        {tasks.map((task) => (
+          <label htmlFor="taskTrashCreation">
+            <li className={!task.completed ? 'task-item' : 'task-item--done'}>
```

Comment:
You should ad a key tag those are used, in React, to identify which items in the list are changed, updated, or deleted!